### PR TITLE
[FLINK-16067][table] Forward Calcite exception when parsing a sql query

### PR DESCRIPTION
**Pull request by dawidwys**
Minor change. Forwarding Calcite exception when parsing a sql query

### CODE REVIEW

**Comment by dawidwys** [comment]
cc @fhueske

**Comment by fhueske** [comment]
Thanks for the fix @dawidwys!
The error details are correctly propagated again :-)

+1 to merge.

File: flink-table/flink-table-planner/src/main/java/org/apache/flink/table/calcite/CalciteParser.java [hunk 1]
```diff
@@ -47,7 +47,7 @@ public SqlNode parse(String sql) {
 			SqlParser parser = SqlParser.create(sql, config);
 			return parser.parseStmt();
 		} catch (SqlParseException e) {
-			throw new SqlParserException("SQL parse failed. " + e.getMessage());
+			throw new SqlParserException("SQL parse failed." + e.getMessage(), e);
```
Comment (libenchao) [inline flink-table/flink-table-planner/src/main/java/org/apache/flink/table/calcite/CalciteParser.java]: seems removing a space by mistake

Comment (dawidwys) [reply flink-table/flink-table-planner/src/main/java/org/apache/flink/table/calcite/CalciteParser.java]: Good catch @libenchao . I fixed it while merging.

Comment (dawidwys) [reply flink-table/flink-table-planner/src/main/java/org/apache/flink/table/calcite/CalciteParser.java]: Good catch @libenchao. Thank you. I fixed it while merging.

**Pull request closed without merging**